### PR TITLE
libsyntax: miscellaneous cleanup

### PR DESCRIPTION
**Pull request by jseyfried**
**syntax-[breaking-change] cc #31645**
Miscellaneous low priority cleanup in `libsyntax` for the next breaking batch.
r? @nrc

### CODE REVIEW

**Comment by jseyfried** [comment]
cc @Manishearth 
I don't know if these cleanups are likely to cause breakage for extensions in the wild.

**Comment by nrc** [comment]
r=me, could probably land early doesn't look like it will break much, but I guess we may as well wait for a batch if there is no urgency.

cc @cgswords

**Comment by Manishearth** [comment]
This probably will break aster.

File: src/libsyntax/ext/expand.rs [hunk 1]
```diff
@@ -87,19 +86,18 @@ impl MacroGenerable for Option<P<ast::Expr>> {
...
-            expand_mac_invoc(mac, None, expr.attrs.into_attr_vec(), expr.span, fld)
+            return expand_mac_invoc(mac, None, expr.attrs.into_attr_vec(), expr.span, fld);
         }
 
         ast::ExprKind::While(cond, body, opt_ident) => {
             let cond = fld.fold_expr(cond);
             let (body, opt_ident) = expand_loop_block(body, opt_ident, fld);
-            fld.cx.expr(expr.span, ast::ExprKind::While(cond, body, opt_ident))
-                .with_attrs(fold_thin_attrs(expr.attrs, fld))
+            expr.node = ast::ExprKind::While(cond, body, opt_ident);
```
Comment (jseyfried) [inline src/libsyntax/ext/expand.rs]: `fold_thin_attrs` became the identity after [this commit](https://github.com/rust-lang/rust/pull/33706/commits/25c733360b1cddfc0b129680a9220d1d5097483d).

**Comment by jseyfried** [comment]
Rebased and added 3 new commits (r? @nrc)

**Comment by nrc** [comment]
r=me

**Pull request merged**